docs/website/docs/general-usage/credentials/configuration.md

```diff
@@ -0,0 +1,422 @@
+
+```python
+@dlt.source
+def google_sheets(
+    spreadsheet_id,
+    tab_names=dlt.config.value,
+    credentials=dlt.secrets.value,
+    only_strings=False
+):
+    sheets = build('sheets', 'v4', credentials=Services.from_json(credentials))
+    tabs = []
+    for tab_name in tab_names:
+        data = sheets.get(spreadsheet_id, tab_name).execute().values()
+        tabs.append(dlt.resource(data, name=tab_name))
+    return tabs
+```
+### Wrong approach
+```python
+# WRONG!:
+# provide all values directly - wrong but possible.
+# secret values should never be present in the code!
+data_source = google_sheets(
+    "23029402349032049",
+    ["tab1", "tab2"],
+    credentials={"private_key": ""}
+)
+```
+:::caution
+Be careful not to put your credentials directly in the code.
+:::
+
+### Right approach
+Provide config values directly and secrets via automatic [injection mechanism](#injection-mechanism).
+`credentials` value will be injected by the `@source` decorator,
+`spreadsheet_id` and `tab_names` take values from the arguments below.
+```python
+# `only_strings` will be injected by the source decorator or will get the default value False
+data_source = google_sheets("23029402349032049", ["tab1", "tab2"])
+```
+or pass everything via configuration:
+```python
+data_source = google_sheets()
+```
+
+In the `google_sheets()` source we can see that some arguments have default values as  `dlt.secrets.value` and `dlt.config.value`.
+
+`dlt.secrets.value` and `dlt.config.value` are instances of classes that provide
+dictionary-like access to configuration values and secrets, respectively.
+These objects allow for convenient retrieval and modification of configuration
+values and secrets used by the application.
+
+`config` deals with non-sensitive configuration data, while `secrets` handles sensitive information
+like passwords, API keys, and other confidential data.
+
+We use these values to set secrets and configurations via:
+- [toml files](config_providers#toml-provider) (secrets.toml & config.toml):
+  ```toml
+  # google sheet credentials
+  [sources.google_sheets.credentials]
+  client_email = <client_email from services.json>
+  private_key = <private_key from services.json>
+  project_id = <project_id from services json>
+  ```
+  Read more about [toml layouts](#secret-and-config-values-layout-and-name-lookup).
+- [Environment Variables](config_providers#environment-provider):
+  ```python
+  SOURCES__GOOGLE_SHEETS__CREDENTIALS__CLIENT_EMAIL
+  SOURCES__GOOGLE_SHEETS__CREDENTIALS__PRIVATE_KEY
+  SOURCES__GOOGLE_SHEETS__CREDENTIALS__PROJECT_ID
+  ```
+
+:::caution
+**[Toml provider](config_providers#toml-provider) always loads those files from `.dlt` folder** which is looked relative to the
+**current Working Directory**.
+:::
+
+
+### Add typing to your sources and resources
+
+You should type your function signatures!
```

Comment:
Let's tone down here a bit ;) 
(I know that this is from the "technical" docs and they less public). For public docs consider  adopting a more suggesting/pragmatic tone.

Reply:
"We recommend to add types to your function signatures. Doing so provides several benefits:..."